bindings/cs/MongoDB.Crypt/LibraryLoader.cs

```diff
@@ -0,0 +1,244 @@
+            // TODO - .NET Standard 2.0
+//            Trace.WriteLine("Base Path: " + basepath)
+
+#if NETSTANDARD1_5
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                string[] suffixPaths = new[]
+                {
+                    "../../native/osx/",
+                    string.Empty
+                };
+                string path = FindLibrary(candidatePaths, suffixPaths, "libmongocrypt.dylib");
+                _loader = new DarwinLibrary(path);
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                string[] suffixPaths = new[]
+                {
+                    "../../native/linux/",
+                    string.Empty
+                };
+                string path = FindLibrary(candidatePaths, suffixPaths, "libmongocrypt.so");
+                _loader = new LinuxLibrary(path);
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+
+#endif
+            {
+                string[] suffixPaths = new[]
+                {
+                    @"..\..\native\windows\",
+                    string.Empty
+                };
+                string path = FindLibrary(candidatePaths, suffixPaths, "mongocrypt.dll");
+                _loader = new WindowsLibrary(path);
+            }
+
+        }
+
+        private string FindLibrary(IList<string> basePaths, string[] suffixPaths, string library)
+        {
+            var candidates = new List<string>();
+            foreach (var basePath in basePaths)
+            {
+                foreach (var suffix in suffixPaths)
+                {
+                    string path = Path.Combine(basePath, suffix, library);
+                    if (File.Exists(path))
+                    {
+                        // TODO - .NET Standard 2.0
+                        //Trace.WriteLine("Load path: " + path);
+                        return path;
+                    }
+                    candidates.Add(path);
+                }
+            }
+
+            throw new FileNotFoundException("Could not find: " + library + " --\n Tried: " + string.Join(",", candidates));
+        }
+
+        public T GetFunction<T>(string name)
+        {
+            IntPtr ptr = _loader.GetFunction(name);
+            if (ptr == IntPtr.Zero)
+            {
+                throw new FunctionNotFoundException(name);
+            }
+
+            return Marshal.GetDelegateForFunctionPointer<T>(ptr);
+
+        }
+
+        public class FunctionNotFoundException : Exception
+        {
+            public FunctionNotFoundException(string message) : base(message) { }
+        }
+
+        private interface ISharedLibraryLoader
+        {
+            IntPtr GetFunction(string name);
+        }
+
+#if NETSTANDARD1_5
+        /// <summary>
+        /// macOS Dynamic Library loader using dlsym
+        /// </summary>
+        private class DarwinLibrary : ISharedLibraryLoader

```

Comment:
The naming is confusing.

Either the class should be named `DarwinLibraryLoader` or the interface should be named `ISharedLibrary`.

But maybe leave alone given time constraints.

Reply:
Fixed